python/ray/tests/test_resource_demand_scheduler.py

```diff
@@ -316,11 +359,62 @@ def test_calculate_node_resources():
     # requires 4 p2.8xls (only 3 are in cluster/pending)
     demands = [{"GPU": 8}] * (len(utilizations) + 2)
     to_launch = scheduler.get_nodes_to_launch(nodes, pending_nodes, demands,
-                                              utilizations)
+                                              utilizations, [])
 
     assert to_launch == {"p2.8xlarge": 1}
 
 
+def test_placement_group_scaling():
+    provider = MockProvider()
+    scheduler = ResourceDemandScheduler(provider, TYPES_A, 10)
+
+    provider.create_node({}, {TAG_RAY_USER_NODE_TYPE: "p2.8xlarge"}, 2)
+    # At this point our cluster has 2 p2.8xlarge instances (16 GPUs) and is
+    # fully idle.
+    nodes = provider.non_terminated_nodes({})
+
+    resource_demands = [{"GPU": 4}] * 2
+    placement_group_load = [
+        # Requires a new node (only uses 2 GPUs on it though).
+        PlacementGroupTableData(
+            state=PlacementGroupTableData.PENDING,
+            strategy=PlacementStrategy.STRICT_SPREAD,
```

Comment:
Can we also add another test that has multiple STRICT_SPREAD groups? I want to ensure that they are properly packed and we don't launch more nodes than needed